src/app/dashboard/chat/[uuid]/page.tsx

```
@@ -13,29 +13,94 @@ import {
   const [modelId, setModelId] = React.useState<string | undefined>(undefined);
   const [style, setStyle] = React.useState<string>('1');
+  const [reasoningEffort, setReasoningEffort] = React.useState<'low' | 'high'>('low');
 
   // Load chat settings from backend
   const { data: chatSettings, refetch: refetchSettings } = api.chat.getChatSettings.useQuery({ chatId: uuid }, { enabled: !!uuid });
   const updateSettings = api.chat.updateChatSettings.useMutation();
 
+  const [characterStats, setCharacterStats] = React.useState({ totalChar: 0, aiCharTotal: 0, userCharTotal: 0 });
+
+  // Memoize all props for ChatUI to prevent remounts
+  const chatUIProps = React.useMemo(() => {
```

Biome (1.9.4)

[error] This hook specifies more dependencies than necessary: setIsRenaming, setCharacterStats

This dependency can be removed from the list.

This dependency can be removed from the list.


(lint/correctness/useExhaustiveDependencies)